### PR TITLE
Implemented BaseClip, videoClip, photoClip, and finalVideo Classes

### PR DESCRIPTION
**Pull request by TheParodicts**
Implemented `BaseClip` Class to establish inheritance for all other specific clips to be used in our software. This includes:

1. Storing the specific type of clip inside a CompositeVideoClip object.

2. Cropping

3. Resizing

4. Adding Text

The following classes can be found in the `ClipClasses.py` file

Implemented `videoClip` and `photoClip` classes that inherit from `BaseClip`, so all that's needed is a modified initialization.

Implemented a finalVideo class that, in theory, is able to concatenate any clip object, as long as it follows the same structure we're using for videoClip.

It's main features are:

- `concatenate_clip(clip)`
 method that allows you to add a clip to the end of the existing finalVideo

- `_add_clip(clip)`
method that allows you to add your own clip to the finalVideo's self.clip. This allows you to add your video on top of any existing part of the video (should only be used after initializing the self.clip with something by using  `start_clip(clip)`

- `_start_clip(clip)`
method to initialize the finalVideo's self.clip to a compositeVideoClip of only one clip. Used primarily for the `build_clip()` method, but can be used outside if you know what you're doing.

### CODE REVIEW

**Review by lexdyel-mendez — APPROVED** [review]

File: animations/finalVideo.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+class finalVideo:
+
+    def __init__(self):
+        self.clip = None
+        self.cliplist = []
+        self.duration = 0
+
+    def start_clip(self, clips):
+        """
+        Slices a video file between two time frames to create a gif
```
Comment (ChristianPerez34) [inline animations/finalVideo.py]: finalVideo shouldn't focus on just creating GIF files. Should be able to create .mp4 as well

File: animations/finalVideo.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                raise AttributeError("Invalid Aspect Ratio specified: '" + str(aspectRatio) + "'")
+
+        # If no preset ratio was selected, use other crop parameters.
+        else:
+            self.clip = self.clip.crop(x1=x1, y1=y1, x2=x2, y2=y2,
+                                       width=width, height=height,
+                                       x_center=x_center, y_center=y_center)
+
+    def writeClip(self, output):
+        # self.clip = self.clip.fx(concatenate([self.clip, self.clip.fx(vfx.time_mirror)]))
```
Comment (ChristianPerez34) [inline animations/finalVideo.py]: Remove comments

File: animations/videoClip.py [hunk 3]
```diff
@@ -112,7 +120,8 @@ def writeClip(self, output):
         self.clip.audio.reader.close_proc()
 
 
-    def add_text(self, text, font_size, color, font, interline, pos, duration):
-        text = TextClip(text, fontsize=font_size, color=color,
-                        font=font, interline=interline).set_pos(pos).set_duration(duration)
-        self.clip = CompositeVideoClip([self.clip, text])
+    # We shouldn't add text here.
```
Comment (ChristianPerez34) [inline animations/videoClip.py]: Remove comments

File: animations/finalVideo.py (hunk 1 above)
Comment (TheParodicts) [reply animations/finalVideo.py]: That was an error from leaving the old comment from the old method. Will update in next push.

**Review by alejoreyes96 — APPROVED** [review]
Organized classes will be very useful when creating different features. Must have

File: animations/ClipClasses.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+class photoClip(BaseClip):
+    def __init__(self, image, duration=5):
+        BaseClip.__init__(self, ImageClip(img=image, duration=duration))
+
+
+class finalVideo(BaseClip):
+
+    def __init__(self):
+        self.clip = None
```
Comment (ChristianPerez34) [inline animations/ClipClasses.py:19]: Why doesn't this one inherit from ```BaseClip```?

**Review by lexdyel-mendez — APPROVED** [review]

File: animations/ClipClasses.py (hunk 4 above)
Comment (TheParodicts) [reply animations/ClipClasses.py:19]: It does, but I didn't want to initialize it the same way the others do. 

For this one, you create the object empty and add in clips that are done.

**Review by ChristianPerez34 — COMMENTED** [review]
Resolve conflicts

**Pull request merged**